### ISSUE TITLE
access missing api’s

### ISSUE THREAD

**Issue by mdahlgrengadd** [godotengine/godot_headers#51]
Sorry for asking, but Is there a way through gdnative to access AudioStreamPlayer’s _mix_to_bus, or is it module only?... Or in general: is it possible to access everything in Godot also in gdnative?

**Comment by karroffel** [comment]
No need to be sorry for asking :slightly_smiling_face: 

The headers here contain mostly "just the core API", so bindings for `Vector2` and `Transform` and the likes.

There are a few calls needed to call into other things, which are the
- `godot_get_class_constructor` which returns a function to construct an object of that type
- `godot_global_get_singleton` which returns the singleton object (for example `Input` and `Engine`)
- `godot_method_bind_get_method` which returns a "binding" to a method of a class. You can use this binding to call the method later on using the following two ways
- `godot_method_bind_ptrcall` calls the method *directly* via pointer and requires the user to set up an array of parameters which contain pointers to the argument.
- `godot_method_bind_call` calls the method the same way GDScript would, adding in default parameters, converting arguments and other "convenience" things, but this is slower than `ptrcall`

This is all very low-level, which is why language bindings generally give you a way to auto-generate those calls and "copy" the Godot API using actual classes/types.

-----

> or is it module only?

> is it possible to access everything in Godot also in gdnative?

GDNative generally only has access to everything that GDScript would have access to as well, it has to go through the `ClassDB` bindings (one exception is "hardcoded" GDNative extensions, for example for networking or VR)

So in short: most stuff should be accessible, but it's a bit more limited that modules, you have to see for the functionality you need yourself.

**Comment by mdahlgrengadd** [comment]
Friendly and informative, thanks 👍
